src/components/Footer/Footer.tsx

```diff
@@ -0,0 +1,69 @@
+import classNames from 'classnames';
+import { SortByStatus } from '../../types/SortByStatus';
+
+type Props = {
+  sortBy: SortByStatus,
+  numberActiveTodos: number,
+  onChangeSortBy: (sortBy: SortByStatus) => void,
+  hasCompletedTodo: boolean,
+};
+
+export const Footer: React.FC<Props> = ({
+  sortBy,
+  numberActiveTodos,
+  onChangeSortBy,
+  hasCompletedTodo,
+}) => {
+  return (
+    <footer className="todoapp__footer">
+      <span className="todo-count">
+        {numberActiveTodos}
+        {' '}
+        items left
+      </span>
+
+      {/* Active filter should have a 'selected' class */}
+      <nav className="filter">
+        <a
+          href="#/"
+          className={classNames(
+            'filter__link',
+            { selected: sortBy === SortByStatus.All },
+          )}
+          onClick={() => onChangeSortBy(SortByStatus.All)}
+        >
+          All
+        </a>
+
+        <a
+          href="#/"
+          className={classNames(
+            'filter__link',
+            { selected: sortBy === SortByStatus.Active },
+          )}
+          onClick={() => onChangeSortBy(SortByStatus.Active)}
+        >
+          Active
+        </a>
+
+        <a
+          href="#/"
+          className={classNames(
+            'filter__link',
+            { selected: sortBy === SortByStatus.Completed },
+          )}
+          onClick={() => onChangeSortBy(SortByStatus.Completed)}
+        >
+          Completed
+        </a>
+      </nav>
+
+      {/* don't show this button if there are no completed todos */}
+      {hasCompletedTodo && (
```

Comment:
I'd recommend using `visibility: hidden` instead of conditional rendering to keep the place for the button in the layout and don't move the filter to the right side if there are no completed todos
P.S. You may fix it in the next stage of app (next task)